daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/MatchingStringModule.scala

```diff
@@ -37,7 +37,10 @@ object MatchingStringModule extends (String => StringModule) {
     private val pattern = regex.pattern
 
     override def fromString(s: String): Either[String, T] =
-      Either.cond(pattern.matcher(s).matches(), s, s"""string "$s" does not match regex "$regex"""")
+      Either.cond(
+        pattern.matcher(s).matches(),
+        s.intern(),
```

Comment:
Also, I _believe_ that `intern` no longer prevents the receiver from ever being GCed, but perhaps we should confirm that?